dali/pipeline/operator/sequence_operator.h

```diff
@@ -406,6 +497,37 @@ class SequenceOperator : public Operator<Backend> {
     }
   }
 
+  void SetupExpandedInputs(const workspace_t<Backend> &ws) {
+    assert(expanded_.NumInput() == 0);
+    for (int input_idx = 0; input_idx < ws.NumInput(); input_idx++) {
+      SetupExpandedInput(ws, input_idx);
+    }
+    assert(expanded_.NumInput() == ws.NumInput());
+  }
+
+  void SetupExpandedOutputs(const workspace_t<Backend> &ws) {
+    assert(expanded_.NumOutput() == 0);
+    for (int output_idx = 0; output_idx < ws.NumOutput(); output_idx++) {
+      SetupExpandedOutput(ws, output_idx);
+    }
+    assert(expanded_.NumOutput() == ws.NumOutput());
+  }
+
+  void SetupExpandedArguments(const workspace_t<Backend> &ws) {
+    for (const auto &arg_input : ws) {
+      auto &shared_tvec = arg_input.second.tvec;
+      assert(shared_tvec);
+      SetupExpandedArgument(ws, arg_input.first, *shared_tvec);
+    }
+  }
+
+  void SetupExpandedWorkspace(const workspace_t<Backend> &ws) {
```

Comment:
Since it's only run once, how about 
```suggestion
  void InitializeExpandedWorkspace(const workspace_t<Backend> &ws) {
```